src/jit/codegencommon.cpp

```diff
@@ -3250,11 +3249,6 @@ void CodeGen::genGCWriteBarrier(GenTree* tgt, GCInfo::WriteBarrierForm wbf)
             unsigned lclNum = 0;
             if (lcl->gtOper == GT_LCL_VAR)
                 lclNum = lcl->gtLclVarCommon.gtLclNum;
```

Comment:
Since the case below is now gone, this should have an `assert(tgtAddr->gtOper == GT_LCL_VAR);`

Reply:
Instead of an explicit assert, `gtLclVarCommon` should be replaced with `AsLclVar` that provides the assert for free. The whole thing should be `unsigned lclNum = lcl->AsLclVar()->GetLclNum();`